### ISSUE TITLE
[LTX-2-FFT] A basic question

### ISSUE THREAD

**Issue by dorpxam** [AkaneTendo25/musubi-tuner#29]
I don't know if the repo is ready to test, but I order to prepare one or more test dataset, my question is maybe naive. Technically, the FFT is a fine-tuning of the model himself, so training directly on the model. The amount of data in the dataset change and are not limited (in principle).

Do you have some recommendation about that point. I will launch some heavy local captioner for the videos/images:

- VIDEO: `DAMO-NLP-SG/VideoLLaMA3-7B` (coupled with `deepseek-r1:14b` for caption enhancement)
- PHOTO: `stepfun-ai/Step3-VL-10B` (the abliterated version `huihui-ai/Huihui-Step3-VL-10B-abliterated`)

Both are GPU consuming. Because `VideoLLaMA3-7B` hallucinate sometimes, I produce 8 captions on each video and send the JSON (using a RAG) to `deepseek-r1:14b` in order to generate a final caption. That works and allow to eliminate the hallucination, but this is time consuming. `Step3-VL-10B` is a hell, between 3/4 minutes to 6/7 minutes by pictures (high resolution), but the prompt adherence is exceptional and the caption quality too. No hallucination, a high level of details, great !

So, I don't want to waste too much time generating hundreds and hundreds of prompts on a dataset that will ultimately be limited. That's where I need your expert advice! 👍 

Thank you for everything!

**Comment by AkaneTendo25** [comment]
I'll test it locally first on a small dataset with a high learning rate, just to make sure the pipeline works properly and that the model actually learns. But the speed on my RTX 3090 is too slow for local-only training on a full dataset.

**Comment by dorpxam** [comment]
About [stepfun-ai/Step3-VL-10B](https://huggingface.co/stepfun-ai/Step3-VL-10B), let me encourage you to use the abliterated version [huihui-ai/Huihui-Step3-VL-10B-abliterated](https://huggingface.co/huihui-ai/Huihui-Step3-VL-10B-abliterated). Not only for NSFW content, but because the abliterated version never reply "_I can’t describe or analyze this image._".

For example, captioning a 36 images set, 9 images was rejected by the official version with a chinese message. Probably the complexity of the photos (even in 1024 x 1536).

<img width="1282" height="567" alt="Image" src="https://github.com/user-attachments/assets/6ce18a26-c0a3-4675-b99c-64315c1e038c" />

The abliterated version works great, even a little faster (here, depending of the complexity of the picture: 101 to 233 seconds)

<img width="2560" height="1392" alt="Image" src="https://github.com/user-attachments/assets/ce7e73a4-63a6-44d7-85b6-a4cf78c77093" />

But produce a almost "ready-to-use" JSON:

```json
{ "image_path": "001-009.png", "caption": "Style: realistic with warm, directional lighting, reminiscent of historical drama portraits. A middle-aged man with a neatly trimmed side-parted light brown hair, a short light brown beard, and subtle freckles across his nose sits in a dimly lit stone chamber. He wears a rich brown leather tunic with fur trim along the shoulders and cuffs, a fur-lined shoulder mantle (a mix of light gray and brown fur), and a leather bracer on his left forearm. His right hand rests on the pommel of a dark wooden-handled sword with a golden guard. Soft overhead light highlights the texture of the leather and fur, casting gentle shadows on his face and the sword. He shifts slightly, adjusting the grip on the sword, his gaze steady and intense as if surveying the chamber. Ambient sounds include the faint echo of his own breath, the subtle rustle of fur against leather, and a distant crackle of a fire in the chamber." }
{ "image_path": "001-016.png", "caption": "Style: realistic with warm, dramatic lighting. In a medium shot, a man in his late 20s with long dark braided hair and a thick dark beard sits on a wooden stool. He wears a rich red velvet chest piece with small silver studs, a thick fur-lined cloak draped over his shoulder, and a brown leather belt. His left hand holds a polished metal helmet with a crossguard, while his right hand grips a long sword with a wooden hilt. Behind him, a large ornate round shield with intricate gold patterns leans against a dark wooden wall, and a slender sword stands vertically nearby. Deep brown drapery forms the background, illuminated by soft warm overhead light that casts gentle shadows on his face and clothing. The man gazes directly at the viewer, his expression calm and contemplative. As he shifts slightly, adjusting the helmet on his knee, the fur of his cloak catches a glint of light. Ambient sounds include the faint rustle of fabric from his movement, the distant echo of a lute strumming a slow melody, and the subtle creak of the wooden stool beneath his weight." }
{ "image_path": "001-017.png", "caption": "Style: realistic with dramatic chiaroscuro lighting. In a medium shot, a middle-aged man with a thick brown beard and short brown hair stands against a misty battlefield backdrop. He wears a rich brown robe with gold floral patterns, a fur-lined cloak with a thick white fur collar, and brown leather gloves. A sword hangs from his belt, alongside a leather pouch. He is adjusting the fur of his cloak with one hand, his gaze fixed on the distant smoky horizon where faint silhouettes of soldiers move. The background features a blurred onion-domed church, with soft light casting deep shadows and highlighting the texture of his robe and the fur. Ambient sounds include the gentle rustle of wind through the cloak, distant arrows whizzing past, and a low, resonant church bell tolling in the distance. His steady breathing is audible, syncing with the subtle shift of his hand as he smooths the fur." }
{ "image_path": "001-018.png", "caption": "Style: realistic with cinematic lighting. A middle-aged man with a shaved head and a thick brown beard sits in a poised stance, wearing a chainmail tunic over a brown leather glove on each hand. He holds a long, ornate sword in his left hand and a polished metal helmet in his right. His lower half is clad in a rich red robe that drapes to his knees, with a light beige undergarment visible beneath. The background is a dark, smoky atmosphere with soft, diffused light highlighting the texture of his chainmail and the sheen of the helmet. As he gently shifts his weight, the chainmail rustles faintly, and the helmet tilts slightly in his grasp. Ambient sounds include the subtle creak of leather from his gloves, the soft metallic clink of the helmet against his palm, and a distant, low rumble of wind through the smoky backdrop. His breath is steady, exhaled softly, adding a quiet rhythm to the scene." }
{ "image_path": "002-011.png", "caption": "Style: moody cinematic portrait with soft directional lighting. A woman with long, wavy dark hair wears an ornate silver crown encrusted with small gemstones. She sits in a relaxed pose, her hands gently clasped in front of her chest, fingers adorned with delicate gold rings. Her attire is a rich dark brown gown with intricate gold embroidery along the sleeves and a fitted bodice, accented by a sheer red panel on her left shoulder and a circular brooch at the neckline. The background is a deep, gradient gray that fades from dark at the edges to a slightly lighter tone behind her, emphasizing her silhouette. Soft ambient sound of a distant harp melody plays, with subtle rustling of her hair and the faint clink of her rings as she breathes. Her gaze lifts slightly toward the upper right, conveying quiet contemplation." }
{ "image_path": "002-012.png", "caption": "Style: realistic with soft studio lighting, reminiscent of 19th-century portraiture. A young girl in early childhood sits gracefully on a dark velvet chair, her long, wavy golden-brown hair tied in a large, loose braid with strands cascading naturally around her shoulders. She wears a delicate pearl headband, a cream-colored sheer blouse with lace cuffs, a dark gray bodice adorned with intricate gold embroidery and trim, and a light gray skirt with gold accents, the front panel of the skirt a deep maroon. Her hands are gently clasped in her lap, holding a turquoise beaded bracelet; she tilts her head slightly, gazing directly at the viewer with a calm, contemplative expression. The background is a dark, muted gradient that fades from deep charcoal at the edges to a soft black near her, ensuring the subject remains the focal point. Ambient sounds fill the space—subtle rustle of her sheer blouse with each slight breath, a faint distant chime echoing softly, and the gentle rise and fall of her chest as she breathes steadily, creating a serene atmosphere." }
{ "image_path": "002-015.png", "caption": "Style: realistic with soft chiaroscuro lighting. A medium shot centers a woman wearing a white headscarf with a darkly patterned band around its crown. She dons a deep blue gown adorned with fine gold embroidery along the edges, layered over a pale blue long-sleeve shirt with ruffled cuffs. In her right hand, she holds a dark beaded necklace that catches subtle light reflections. The background is a smooth gradient shifting from dark gray at the top to black at the bottom, which emphasizes the subject. The woman turns her head slightly toward the viewer, her gaze calm and direct, while her left hand rests gently on the gown’s side seam. Soft ambient sounds accompany the scene: a faint harp glissando plays in the distance, and the gentle rustle of her headscarf with each subtle breath. The lighting highlights the texture of the fabric and the delicate pattern on her headband, creating a serene, timeless portrait atmosphere." }
{ "image_path": "002-017.png", "caption": "Style: realistic with warm, soft lighting, reminiscent of 19th-century portraiture. A medium shot captures a woman in her late 20s with chestnut-brown hair styled in an elegant updo adorned with pink and yellow flowers. She wears a flowing gown with a floral pattern in gold and olive tones, featuring ruffled sleeves, a lace-trimmed neckline, and a white oval pendant necklace. Her right hand gently holds a white feather that quivers slightly, while her left hand rests lightly on the gown’s bodice. To her left stands an antique candelabra with a flickering candle casting a warm amber glow, and a small table holds a white porcelain teacup with a floral design. To the right, a three-tiered glass dessert stand displays fresh strawberries, pastel macarons, and pink roses. The background is a deep brown gradient, with soft side lighting from above-left highlighting the textures of the fabric and the delicate details of the flowers. Ambient sounds include the faint crackle of the candle flame, a subtle background melody of a classical piano piece, and the gentle clink of porcelain as the dessert stand shifts slightly. The woman gazes toward the right with a soft smile, her lips slightly parted as if breathing in the quiet moment. A faint breeze moves the dark drapery behind her, producing a whisper of fabric rustling." }
{ "image_path": "002-018.png", "caption": "Style: cinematic dark fantasy with warm, diffused lighting. A middle-aged woman with dark, wavy hair wears an ornate crown encrusted with small gemstones, long dangling earrings, and a heavy gold necklace with a large pendant. She is dressed in a rich burgundy off-shoulder gown with a thick golden fur trim on the shoulder, a metallic corset, and a flowing burgundy skirt. She sits slightly forward, holding a wooden-handled battle axe in her right hand—the blade is dark metal with intricate carvings—while her left hand rests on her knee, fingers adorned with rings. The background features a misty, dark sky with swirling clouds and faint silhouettes of ancient stone crosses in the distance; the mist drifts slowly behind her. She turns her head slightly left, gazing beyond the frame with quiet authority, as a gentle breeze lifts the fur trim and makes her skirt flutter. Ambient sounds include a low pipe organ melody, the soft rustle of her skirt and fur with each subtle movement, the faint echo of wind through the mist, and her steady breath—accompanied by a faint metallic click as the axe shifts in her grip." }
```

The only constraint of the abliterated version is that you need to clone the huggingface directory locally and hack two of the python's code file, here is the trick to avoid the error:

In `modeling_step_vl.py`, add a dot before `configuration_step_vl` & `vision_encoder` (lines 32 & 33)
```python
from .configuration_step_vl import StepRoboticsConfig
from .vision_encoder import StepRoboticsVisionEncoder
```
In `vision_encoder.py`, add a dot before `configuration_step_vl` (lines 9)
```python
from .configuration_step_vl import StepRoboticsVisionEncoderConfig
```

That's all, and works great, I'm impressed by this model. Slow but great!

## EDIT

Ah yes, I use the same system prompt that the one used by Kijai in the ComfyUI's node `TextGenerateLTX2Prompt` and a blank user prompt (only the photos as input).

Here is the system prompt:

```python
LTX2_T2V_SYSTEM_PROMPT = """\
You are a Creative Assistant. Given a user's raw input prompt describing a scene or concept, expand it into a detailed video generation prompt with specific visuals and integrated audio to guide a text-to-video model.
#### Guidelines
- Strictly follow all aspects of the user's raw input: include every element requested (style, visuals, motions, actions, camera movement, audio).
    - If the input is vague, invent concrete details: lighting, textures, materials, scene settings, etc.
        - For characters: describe gender, clothing, hair, expressions. DO NOT invent unrequested characters.
- Use active language: present-progressive verbs ("is walking," "speaking"). If no action specified, describe natural movements.
- Maintain chronological flow: use temporal connectors ("as," "then," "while").
- Audio layer: Describe complete soundscape (background audio, ambient sounds, SFX, speech/music when requested). Integrate sounds chronologically alongside actions. Be specific (e.g., "soft footsteps on tile"), not vague (e.g., "ambient sound is present").
- Speech (only when requested):
    - For ANY speech-related input (talking, conversation, singing, etc.), ALWAYS include exact words in quotes with voice characteristics (e.g., "The man says in an excited voice: 'You won't believe what I just saw!'").
    - Specify language if not English and accent if relevant.
- Style: Include visual style at the beginning: "Style: <style>, <rest of prompt>." Default to cinematic-realistic if unspecified. Omit if unclear.
- Visual and audio only: NO non-visual/auditory senses (smell, taste, touch).
- Restrained language: Avoid dramatic/exaggerated terms. Use mild, natural phrasing.
    - Colors: Use plain terms ("red dress"), not intensified ("vibrant blue," "bright red").
    - Lighting: Use neutral descriptions ("soft overhead light"), not harsh ("blinding light").
    - Facial features: Use delicate modifiers for subtle features (i.e., "subtle freckles").

#### Important notes:
- Analyze the user's raw input carefully. In cases of FPV or POV, exclude the description of the subject whose POV is requested.
- Camera motion: DO NOT invent camera motion unless requested by the user.
- Speech: DO NOT modify user-provided character dialogue unless it's a typo.
- No timestamps or cuts: DO NOT use timestamps or describe scene cuts unless explicitly requested.
- Format: DO NOT use phrases like "The scene opens with...". Start directly with Style (optional) and chronological scene description.
- Format: DO NOT start your response with special characters.
- DO NOT invent dialogue unless the user mentions speech/talking/singing/conversation.
- If the user's raw input prompt is highly detailed, chronological and in the requested format: DO NOT make major edits or introduce new elements. Add/enhance audio descriptions if missing.

#### Output Format (Strict):
- Single continuous paragraph in natural language (English).
- NO titles, headings, prefaces, code fences, or Markdown.
- If unsafe/invalid, return original user prompt. Never ask questions or clarifications.

Your output quality is CRITICAL. Generate visually rich, dynamic prompts with integrated audio for high-quality video generation.

#### Example
Input: "A woman at a coffee shop talking on the phone"
Output:
Style: realistic with cinematic lighting. In a medium close-up, a woman in her early 30s with shoulder-length brown hair sits at a small wooden table by the window. She wears a cream-colored turtleneck sweater, holding a white ceramic coffee cup in one hand and a smartphone to her ear with the other. Ambient cafe sounds fill the space—espresso machine hiss, quiet conversations, gentle clinking of cups. The woman listens intently, nodding slightly, then takes a sip of her coffee and sets it down with a soft clink. Her face brightens into a warm smile as she speaks in a clear, friendly voice, 'That sounds perfect! I'd love to meet up this weekend. How about Saturday afternoon?' She laughs softly—a genuine chuckle—and shifts in her chair. Behind her, other patrons move subtly in and out of focus. 'Great, I'll see you then,' she concludes cheerfully, lowering the phone.
"""
```

**Comment by dorpxam** [comment]
Just a little question about your resolution choice (image-only) for the fine-tuning ?

- Did you use multiple resolution ?
- What is your choice (minimum/maximum size) ?

Thanks a lot for your help!

**Comment by AkaneTendo25** [comment]
I haven't decided yet, still experimenting :) But I think 512p is the lower bound.
Multi-resolution is probably fine, but I want to experiment with speed first to understand what the maximum training resolution is without pushing VRAM requirements too high.

**Comment by dorpxam** [comment]
> I haven't decided yet, still experimenting :)

Understand. Me too :)

> But I think 512p is the lower bound.

largest dimension or lowest ? Like 768x512 / 512x768 ?

> Multi-resolution is probably fine, but I want to experiment with speed first to understand what the maximum training resolution is without pushing VRAM requirements too high.

Yes. I don't know exactly how the model works about the resolution, but I probably use a maximum of 1600px (in largest dimension, like 1024x1536). I think it's consistent to preserve the details and not get bogged down with excessively high resolutions that are probably unnecessary in the end. 

Regarding the details, by using very high resolution sources (> 10000px) it is possible to cut precise areas while maintaining perfect resolution (for example, a close-up view of the eyes).

**Comment by AkaneTendo25** [comment]
>> But I think 512p is the lower bound.
> 
> largest dimension or lowest ? Like 768x512 / 512x768 ?

512p in terms of bucketing resolution, i.e. pixel area (512x512 = 262144). This fits resolutions like 768x352 or 640x416 etc.

**Comment by dorpxam** [comment]
> 512p in terms of bucketing resolution, i.e. pixel area

OK. I see. Thank you very much ! 👍
